documentation/src/main/asciidoc/topics/proc_storing_custom_java_objects.adoc

```diff
@@ -0,0 +1,42 @@
+[id='storing_binary_objects-{context}']
+= Storing Java Objects in Binary Format
+Java Hot Rod clients can handle Java objects that represent entities and
+perform marshalling to serialize and deserialize objects into `byte[]` arrays.
+C{plusplus}, C#, and Javascript Hot Rod clients can also handle objects in the
+respective languages.
+
+If you store entries in the cache as marshalled Java objects, you should configure the cache with the MediaType of the marshalled storage.
+
+.Procedure
+
+. Specify the MediaType that matches your marshaller implementation.
++
+* Protostream marshaller: Configure the MediaType as `application/x-protostream`.
+* JBoss marshalling: Configure the MediaType as `application/x-jboss-marshalling`.
+* StringMarshaller: Configure the MediaType as `text/plain`.
++
+[NOTE]
+====
+Alternatively, you can define a global marshaller with the `serialization`
```

Comment:
This is the "user marshaller" as it's only concerned with user key/values. The global marshaller is what we use internally for Infinispan classes.

Reply:
thanks @ryanemerson that's a good clarification. can you please confirm the changes I made?